File: scripts/process_logo.py
"""Generate logo assets from source PNG. Run: python scripts/process_logo.py"""
from pathlib import Path
from PIL import Image

ROOT = Path(__file__).resolve().parents[1]
SOURCE = Path(r"C:\Users\1\Downloads\ChatGPT Image 7 июн. 2026 г., 10_58_40.png")

if not SOURCE.exists():
    raise SystemExit(f"Logo source not found: {SOURCE}")

img = Image.open(SOURCE).convert("RGBA")
size = img.width

compact_bottom = round(size * 0.815)
icon_bottom = round(size * 0.46)

def crop_box(top: int, bottom: int) -> Image.Image:
    return img.crop((0, top, size, bottom))

def fit_square(im: Image.Image, out_w: int, bg=(0, 0, 0, 255)) -> Image.Image:
    if im.mode != "RGBA":
        im = im.convert("RGBA")
    scale = min(out_w / im.width, out_w / im.height)
    nw, nh = max(1, round(im.width * scale)), max(1, round(im.height * scale))
    resized = im.resize((nw, nh), Image.Resampling.LANCZOS)
    canvas = Image.new("RGBA", (out_w, out_w), bg)
    ox, oy = (out_w - nw) // 2, (out_w - nh) // 2
    canvas.paste(resized, (ox, oy), resized)
    return canvas

icon_src = crop_box(0, icon_bottom)
compact_src = crop_box(0, compact_bottom)
full_src = img

web = ROOT / "website" / "assets"
web.mkdir(parents=True, exist_ok=True)

fit_square(full_src, 1024, (0, 0, 0, 255)).convert("RGB").save(web / "logo-full.png")
fit_square(compact_src, 1024, (0, 0, 0, 255)).convert("RGB").save(web / "logo-compact.png")
fit_square(icon_src, 1024, (0, 0, 0, 255)).convert("RGB").save(web / "logo-icon.png")

for w, name in [(512, "icon-512.png"), (32, "favicon-32.png"), (180, "apple-touch-icon.png")]:
    fit_square(icon_src, w, (0, 0, 0, 255)).convert("RGB").save(web / name)

android_res = ROOT / "android" / "app" / "src" / "main" / "res"
densities = {
    "mipmap-mdpi": 108,
    "mipmap-hdpi": 162,
    "mipmap-xhdpi": 216,
    "mipmap-xxhdpi": 324,
    "mipmap-xxxhdpi": 432,
}

for folder, px in densities.items():
    d = android_res / folder
    d.mkdir(parents=True, exist_ok=True)
    fit_square(icon_src, px, (0, 0, 0, 0)).save(d / "ic_launcher_foreground.png")
    fit_square(compact_src, px, (0, 0, 0, 255)).convert("RGB").save(d / "ic_launcher.png")
    fit_square(compact_src, px, (0, 0, 0, 255)).convert("RGB").save(d / "ic_launcher_round.png")

drawable = android_res / "drawable-nodpi"
drawable.mkdir(parents=True, exist_ok=True)
fit_square(compact_src, 512, (0, 0, 0, 255)).convert("RGB").save(drawable / "logo_efir_ai.png")

print("done")

File: scripts/process-logo.mjs
/**
 * Generate logo assets from source PNG.
 * Run: node scripts/process-logo.mjs
 */
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';
import sharp from 'sharp';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.resolve(__dirname, '..');

const SOURCE = process.env.LOGO_SOURCE
  ?? path.join(process.env.USERPROFILE ?? '', 'Downloads', 'ChatGPT Image 7 июн. 2026 г., 10_58_40.png');

if (!fs.existsSync(SOURCE)) {
  console.error('Logo source not found:', SOURCE);
  process.exit(1);
}

const { width, height } = await sharp(SOURCE).metadata();
const size = width ?? 1254;

/** Symbol + ЭФИР AI — без нижнего слогана. */
const compactBottom = Math.round(size * 0.815);
/** Только знак с волнами (верхняя часть). */
const iconBottom = Math.round(size * 0.46);

async function cropExtract(name, top, bottom, outW, outPath) {
  const h = bottom - top;
  await sharp(SOURCE)
    .extract({ left: 0, top, width: size, height: h })
    .resize(outW, outW, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .png()
    .toFile(outPath);
  console.log('wrote', outPath);
}

async function resizeSquare(input, outW, outPath, bg = '#000000') {
  await sharp(input)
    .resize(outW, outW, { fit: 'contain', background: bg })
    .png()
    .toFile(outPath);
  console.log('wrote', outPath);
}

const webAssets = path.join(ROOT, 'website', 'assets');
const androidRes = path.join(ROOT, 'android', 'app', 'src', 'main', 'res');
fs.mkdirSync(webAssets, { recursive: true });

const tmpDir = path.join(ROOT, 'scripts', '.logo-tmp');
fs.mkdirSync(tmpDir, { recursive: true });

const iconTmp = path.join(tmpDir, 'icon.png');
const compactTmp = path.join(tmpDir, 'compact.png');
const fullTmp = path.join(tmpDir, 'full.png');

await cropExtract('icon', 0, iconBottom, 1024, iconTmp);
await cropExtract('compact', 0, compactBottom, 1024, compactTmp);
await sharp(SOURCE).resize(1024, 1024, { fit: 'contain', background: '#000000' }).png().toFile(fullTmp);

// Website
await sharp(fullTmp).toFile(path.join(webAssets, 'logo-full.png'));
await sharp(compactTmp).toFile(path.join(webAssets, 'logo-compact.png'));
await sharp(iconTmp).toFile(path.join(webAssets, 'logo-icon.png'));

for (const [w, name] of [[512, 'icon-512.png'], [32, 'favicon-32.png'], [180, 'apple-touch-icon.png']]) {
  await resizeSquare(iconTmp, w, path.join(webAssets, name));
}

// Android mipmaps — foreground (icon on transparent for adaptive)
const densities = {
  'mipmap-mdpi': 108,
  'mipmap-hdpi': 162,
  'mipmap-xhdpi': 216,
  'mipmap-xxhdpi': 324,
  'mipmap-xxxhdpi': 432,
};

for (const [folder, px] of Object.entries(densities)) {
  const dir = path.join(androidRes, folder);
  fs.mkdirSync(dir, { recursive: true });
  await resizeSquare(iconTmp, px, path.join(dir, 'ic_launcher_foreground.png'), { r: 0, g: 0, b: 0, alpha: 0 });
  await resizeSquare(compactTmp, px, path.join(dir, 'ic_launcher.png'), '#000000');
  await resizeSquare(compactTmp, px, path.join(dir, 'ic_launcher_round.png'), '#000000');
}

// In-app logo drawable (compact wordmark)
const drawableDir = path.join(androidRes, 'drawable-nodpi');
fs.mkdirSync(drawableDir, { recursive: true });
await sharp(iconTmp).toFile(path.join(drawableDir, 'logo_efir_ai.png'));

console.log('done');
